.github/workflows/jsdoc.yml

```diff
@@ -0,0 +1,47 @@
+name: JSDoc
+on:
+  pull_request:
+    paths:
+      # documentation
+      - "docs/**"
+      - "*.md"
+      # changes to this workflow
+      # seems there is no way to get this easily yet https://github.com/actions/runner/issues/853
+      - ".github/workflows/jsdoc.yml"
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.0
+
+env:
+  REALM_DISABLE_ANALYTICS: 1
+jobs:
+  jsdoc:
+    name: JSDoc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 14
+          registry-url: https://registry.npmjs.org/
+      - name: Install npm v7
+        run: npm install -g npm@7
+      # Install the root package (--ignore-scripts to avoid downloading or building the native module)
+      - name: Install root package dependencies
+        run: npm ci --ignore-scripts
+      - name: Install subpackages dependencies
+        run: npm run lerna:bootstrap
+      - name: Run JSDoc documentation build
+        run: npm run jsdoc
+      - name: Configure AWS Credentials
+        if: startsWith(github.ref, 'refs/tags/v')
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.DOCS_S3_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.DOCS_S3_SECRET_KEY }}
+          aws-region: us-west-1
+      - name: Upload docs
+        if: startsWith(github.ref, 'refs/tags/v')
+        run:  aws s3 sync --acl public-read docs/output/realm s3://${{ secrets.DOCS_S3_BUCKET_NAME }}
```

Comment:
```suggestion
        run: aws s3 sync --acl public-read docs/output/realm s3://${{ secrets.DOCS_S3_BUCKET_NAME }}
```

Reply:
I'm wondering if `DOCS_S3_BUCKET_NAME` need to be considered a secret?

Reply:
Also, doesn't destination path need to contain the current package version?

Reply:
Bucket name was easier to configure as a variable. I would need to know the bucket name to hardcode it.
The documentation is already built in a directory named as current package version, so the step uploads it as is.
For example docs/output/realm/10.2.0 is uploaded to 10.2.0